coral/src/app/context-provider/AuthProvider.tsx

```diff
@@ -0,0 +1,22 @@
+import { createContext, ReactNode, useContext } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { AuthUser, getAuth } from "src/domain/auth-user";
+
+// for now, we don't do Authentication on Corals side,
+// so we only have a AuthUser in the context
+const AuthContext = createContext<AuthUser | undefined>(undefined);
+
+const useAuthContext = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const { data: authUser } = useQuery<AuthUser | undefined>(
+    ["user-getAuth-data"],
+    getAuth
+  );
+
+  return (
+    <AuthContext.Provider value={authUser}>{children}</AuthContext.Provider>
+  );
+};
```

Comment:
Should we prevent rendering children already here until `authUser` is successfully resolved?

Reply:
... and maybe even do the redirect here if the query fails?

```
{ error && <Redirect to={{ pathname: "/login", state: { from: location } }} /> }
```

Reply:
>Should we prevent rendering children already here until authUser is successfully resolved?

Yes, that sounds good! Then I probably don't need to have the same check in `Layout`.

> ... and maybe even do the redirect here if the query fails? 

I'll check how that would work out. Maybe in that case we don't need the Error boundary around the AuthContext, if it takes care about the error itself 🤔 🤔 🤔  

Will ping you when I've an update!

Reply:
Updated! Works very nicely I think 👯

Reply:
Actually turns out it did not work nicely :D